Tests failing (on Arch)

**Do you want to request a *feature* or report a *bug*?**   Bug

**What is the current behavior?**

This was first identified when the Arch AUR package for todotxt failed because of the test failures. Another user reported the same failure; I pulled this repo and got the same results: `make test` fails tests.

Reference: https://aur.archlinux.org/packages/todotxt/#comment-817888

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

Ensure no environment variables are set that might interfere with tests:

```
10083 » printenv | grep TODO
10084 » 
```

Start with a clean clone:
```
10081» git clone https://github.com/todotxt/todo.txt-cli
Cloning into 'todo.txt-cli'...
remote: Enumerating objects: 2325, done.
remote: Total 2325 (delta 0), reused 0 (delta 0), pack-reused 2325
Receiving objects: 100% (2325/2325), 5.61 MiB | 1018.00 KiB/s, done.
Resolving deltas: 100% (1399/1399), done.
```

Make and make test the project:
```
10082» cd todo.txt-cli && make && make test
VERSION=2.12.0
make: 'VERSION-FILE' is up to date.
rm -rf tests/test-results "tests/trash directory"*
cd tests && ./t0000-config.sh
* FAIL 1: no config file

    todo.sh > output 2>&1 || test_cmp expect output

*   ok 2: config file (default location 1)
*   ok 3: config file (default location 2)
*   ok 4: config file (default location 3)
* FAIL 5: config file (global config file)

    cp test.cfg "$TODOTXT_GLOBAL_CFG_FILE"
    todo.sh > output;
    test_cmp expect output && test -f used_config &&
        rm -f "$TODOTXT_GLOBAL_CFG_FILE"

*   ok 6: config file (command line)
*   ok 7: config file (env variable)
*   ok 8: config file (minimal)
* failed 2 among 8 test(s)
make: *** [Makefile:96: tests/t0000-config.sh] Error 1
```

**What is the expected behavior?**

Unit tests pass.

**Which versions todo.sh are you using?** N/A

**Which Operating System are you using?** Linux Arch 5.12.12-arch1-1

**Which version of bash are you using?** GNU bash, version 5.1.8(1)-release (x86_64-pc-linux-gnu)

--- a/tests/test-lib.sh
+++ b/tests/test-lib.sh
@@ -693,6 +693,8 @@ cd -P "$test" || exit 1
 # but use something specified by the framework.
 HOME=$(pwd)
 export HOME
+# Unset XDG_CONFIG_HOME as that is used as a config alternative.
+unset XDG_CONFIG_HOME
 
 this_test=${0##*/}
 this_test=${this_test%%-*}

--- a/tests/test-lib.sh
+++ b/tests/test-lib.sh
@@ -695,6 +695,11 @@ HOME=$(pwd)
 export HOME
 # Unset XDG_CONFIG_HOME as that is used as a config alternative.
 unset XDG_CONFIG_HOME
+# User add-ons may override built-in commands; these could have incompatible
+# behavior that makes the tests fail. Avoid picking up user add-ons by
+# explicitly configuring the first default location (which with the redirected
+# HOME lies within the test directory and usually does not exist).
+export TODO_ACTIONS_DIR="$HOME/.todo/actions"
 
 this_test=${0##*/}
 this_test=${this_test%%-*}

